spec.md

```diff
@@ -0,0 +1,954 @@
+
+#### Secrets
+
+The `MachineClass` resource also contains a `secretRef` field that contains a reference to a secret. The keys of this secret are passed in the `Secrets` request field to CMI methods.
+
+The secret can contain sensitive data such as
+- `cloud-credentials` secret data used to authenticate at the provider
+- `cloud-init` scripts used to initialize a new VM. The cloud-init script is expected to contain scripts to initialize the Kubelet and make it join the cluster.
+
+#### Identifying Cluster Machines
+
+To implement certain methods, the plugin should be able to identify all machines associated with a particular Kubernetes cluster. This can be achieved using one/more of the below mentioned ways:
+
+* Names of VMs created by the plugin are prefixed by the cluster ID specified in the [ProviderSpec](#providerspec).
+* VMs created by the plugin are tagged with the special tags like `kubernetes.io/cluster` (for the cluster ID) and `kubernetes.io/role` (for the role), specified in the [ProviderSpec](#providerspec).
+* Mapping `Resource Groups` to individual cluster.
+
+### Identity Service RPC
+
+Identity service RPCs allow a CMI-Client to query a CMI-Plugin for capabilities, health, and other metadata.
+The general flow of the success case MAY be as follows (protos illustrated in YAML for brevity):
+
+1. CMI-Client queries metadata via Identity RPC.
+
+```
+   # CMI-Client --(GetPluginInfo)--> Plugin
+   request:
+   response:
+      name: org.foo.whizbang.super-plugin
+      vendor_version: blue-green
+      manifest:
+        baz: qaz
+```
+
+2. CMI-Client queries available capabilities of the plugin.
+
+```
+   # CMI-Client --(GetPluginCapabilities)--> Plugin
+   request:
+   response:
+     capabilities:
+       - service:
+           type: CONTROLLER_SERVICE
+```
+
+3. CMI-Client queries the readiness of the plugin.
+
+```
+   # CMI-Client --(Probe)--> Plugin
+   request:
+   response: {}
+```
+
+#### `GetPluginInfo`
+
+```protobuf
+message GetPluginInfoRequest {
+    // Intentionally empty.
+}
+
+message GetPluginInfoResponse {
+    // The name MUST follow domain name notation format
+    // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
+    // include the plugin's host company name and the plugin name,
+    // to minimize the possibility of collisions. It MUST be 63
+    // characters or less, beginning and ending with an alphanumeric
+    // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+    // alphanumerics between.
+    // This field is REQUIRED.
+    string name = 1;
+
+    // The version specifies the plugin version
+    // Value of this field is opaque to the CMI-Client.
+    // This field is REQUIRED.
+    string version = 2;
+
+    // manifest contains a map of key-value pairs to pass
+    // any additonal information about the plugin.
+    // Values are opaque to the CMI-Client.
+    // This field is OPTIONAL.
+    map<string, string> manifest = 3;
```

Comment:
The idea is for the manifest to be a map of key-value pairs. This will contain any addtional information on the plugin. E.g.: key = start-time, value = timestamp (xx.xx.xxxx xx:xx:xx IST).

Reply:
Have added an example for the usage.